msbuild/Xamarin.Mac.Tasks.Core/Tasks/DetectSdkLocations.cs

```diff
@@ -0,0 +1,145 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+using Xamarin.MacDev.Tasks;
+using Xamarin.MacDev;
+
+namespace Xamarin.Mac.Tasks
+{
+	public class DetectSdkLocationsTaskBase : Task
+	{
+		#region Inputs
+
+		public string SessionId { get; set; }
+
+		// This is also an input
+		[Output]
+		public string SdkVersion {
+			get; set;
```

Comment:
If you have the file moves (only) in a separate commits, then git should properly detect that and not show that many additions/deletions.

Then in a separate commit you can change things (in this case I see you've changed the name of the class, but I have no idea if you've changed anything else in the file, which seriously complicates reviewing the PR).